510: Add documentation on overriding GraphQL classes (#511)
workfloworchestrator/orchestrator-core: docs/architecture/application/graphql.md

```

By following these examples, you can effectively federate autogenerated types (`subscriptions` and `product blocks`) enabling seamless integration across multiple GraphQL endpoints.

## Overriding Types

Overriding strawberry types can be achieved through various methods. One less desirable approach involves extending classes using class inheritance.
However, this method becomes cumbersome when updating a single class, as it necessitates updating all associated types and their corresponding resolvers, essentially impacting the entire structure.

For instance, consider the scenario of overriding the `CustomerType`. This would entail updating the `SubscriptionInterface`, `ProcessType` and their respective resolvers. Subsequently, due to these modifications, all related types and their resolvers would also require updates, resulting in a tedious and error-prone process.

> To streamline and enhance the update process, a more efficient solution has been devised. This involves using a helper function called `override_class`, specifically designed to override fields within a class. This function takes the base class as well as a list of fields that will replace their counterparts within the class.

Reviewer: 'To streamline and enhance the update process, a more efficient solution has been devised."

Not sue how much these kind of sentences add to the readability. Mostly AI generated noise to mee.

Author: I mean its not great, but found my own sentence worse: `I found a much better way to update classes and because its not a pretty way, I created a helper function to override fields called 'override_class'`

Author: ```suggestion
To enhance the override process, we created a helper function `override_class` to specifically override fields, It takes the base class as well as a list of fields that will replace their counterparts within the class or add new fields.
```
